customized_list_view/models/custom_list.py

```diff
@@ -0,0 +1,139 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import json
+
+from lxml import etree
+
+from odoo import _, api, exceptions, fields, models
+
+
+class CustomListView(models.Model):
+    _name = "custom.list.view"
+    _description = "Custom List View"
+
+    name = fields.Char(required=True)
+    model_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    model_name = fields.Char(compute="_compute_model_name", store=True)
+    list_view_id = fields.Many2one("ir.ui.view", required=True)
+    line_ids = fields.One2many("custom.list.view.line", "custom_list_view_id")
+    original_arch = fields.Text(readonly=True)
+
+    @api.constrains("model_id")
+    def _check_unique_model(self):
+        for rec in self:
+            same_model_recs = self.env["custom.list.view"].search(
+                [
+                    ("model_id", "=", rec.model_id.id),
+                    ("id", "!=", rec.id),
+                ]
+            )
+            if same_model_recs:
+                raise exceptions.ValidationError(
+                    _(
+                        "Only one record per model is allowed. "
+                        "Please modify existing record instead."
+                    )
+                )
+
+    @api.depends("model_id")
+    def _compute_model_name(self):
+        for rec in self:
+            rec.model_name = rec.model_id.model
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res.original_arch = res.list_view_id.arch
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        return res
+
+    def button_apply_changes(self):
+        self.ensure_one()
+        doc = etree.XML(self.original_arch)
+        for mod_line in self.line_ids:
+            target = mod_line.before.name if mod_line.before else mod_line.after.name
+            for node in doc.xpath("/tree/field[@name='%s']" % target):
+                node_string = (
+                    "<field name='%s' widget='%s' optional='%s' string='%s'/>"
+                    % (
+                        mod_line.field_id.name,
+                        mod_line.widget or "",
+                        mod_line.optional or "",
+                        mod_line.label or mod_line.field_id.field_description,
+                    )
+                )
+                new_node = etree.fromstring(node_string)
+                if mod_line.before:
+                    node.addprevious(new_node)
+                else:
+                    node.addnext(new_node)
+        new_arch = etree.tostring(doc, encoding="unicode").replace("\t", "")
+        self.list_view_id.arch = new_arch
+
+    def button_roll_back(self):
+        self.ensure_one()
+        doc = etree.XML(self.original_arch)
+        new_arch = etree.tostring(doc, encoding="unicode").replace("\t", "")
+        self.list_view_id.arch = new_arch
+
+
+class CustomListViewLine(models.Model):
```

Comment:
I think better this class get out of separate file with name `customer_list_view_line.py`.

Reply:
> I think better this class get out of separate file with name `customer_list_view_line.py`.

@ilyasProgrammer What do you think about it?